core/tests/puppeteer-acceptance-tests/puppeteer-testing-utilities/puppeteer-utils.js

```diff
@@ -0,0 +1,228 @@
+   */
+  async signUpNewUser(userName, signInEmail) {
+    await this.signInWithEmail(signInEmail);
+    await this.type('input.e2e-test-username-input', userName);
+    await this.clickOn('input.e2e-test-agree-to-terms-checkbox');
+    await this.page.waitForSelector(
+      'button.e2e-test-register-user:not([disabled])');
+    await this.clickOn(LABEL_FOR_SUBMIT_BUTTON);
+    await this.page.waitForNavigation({waitUntil: 'networkidle0'});
+  }
+
+  /**
+   * Function to reload the current page.
+   */
+  async reloadPage() {
+    await this.page.reload({waitUntil: ['networkidle0', 'domcontentloaded']});
+  }
+
+  /**
+   * The function clicks the element using the text on the button.
+   * @param {string} selector - The text on the button or the CSS selector of
+   * the element to be clicked
+   */
+  async clickOn(selector) {
+    try {
+      const [button] = await this.page.$x(
+        '//' + `*[contains(text(), '${selector}')]`);
+      await button.click();
+    } catch (error) {
+      await this.page.waitForSelector(selector);
+      await this.page.click(selector);
+    }
+  }
+
+  /**
+   * The function to assign a role to a user.
+   * @param {string} username - The username to which role would be assigned.
+   * @param {string} role - The role that would be assigned to the user.
+   */
+  async assignRoleToUser(username, role) {
+    await this.goto(rolesEditorTab);
+    await this.type(roleEditorInputField, username);
+    await this.clickOn(roleEditorButtonSelector);
+    await this.clickOn(addRoleButton);
+    await this.clickOn(rolesSelectDropdown);
+    await this.page.evaluate(async(role) => {
+      const allRoles = document.getElementsByClassName('mat-option-text');
+      for (let i = 0; i < allRoles.length; i++) {
+        if (allRoles[i].innerText.toLowerCase() === role) {
+          allRoles[i].click({waitUntil: 'networkidle0'});
+          return;
+        }
+      }
+      throw new Error('Role ' + role + ' does not exists.');
+    }, role);
+  }
+
+  /**
+   * The function excepts the user to have the given role.
+   * @param {string} username - The username to which role must be assigned.
+   * @param {string} role - The role which must be assigned to the user.
+   */
+  async expectUserToHaveRole(username, role) {
+    const currentPageUrl = this.page.url();
+    await this.goto(rolesEditorTab);
+    await this.type(roleEditorInputField, username);
+    await this.clickOn(roleEditorButtonSelector);
+    await this.page.waitForSelector('div.justify-content-between');
+    await this.page.evaluate((role) => {
+      const userRoles = document.getElementsByClassName(
+        'oppia-user-role-description');
+      for (let i = 0; i < userRoles.length; i++) {
+        if (userRoles[i].innerText.toLowerCase() === role) {
+          return;
+        }
+      }
+      throw new Error('User does not have the ' + role + ' role!');
+    }, role);
+    showMessage('User ' + username + ' has the ' + role + ' role!');
```

Comment:
Why not use formatting strings? Ditto elsewhere in this PR.

Reply:
done everywhere.. but some places we need to break line when it exceeds the line limit of 80